guide/phone_order_delivery_service/phone_order_delivery_service.bal

```diff
@@ -0,0 +1,218 @@
+            done;
+        }
+
+        // Order details
+        newDeliver.customerName = name.toString();
+        newDeliver.address = address.toString();
+        newDeliver.contactNumber = contact.toString();
+        newDeliver.deliveryPhoneName = phoneName.toString();
+
+        // boolean variable to track the availability of a requested phone
+        boolean isPhoneAvailable;
+        // Check whether the requested phone available
+        foreach phone in phoneInventory {
+            if (newDeliver.deliveryPhoneName.equalsIgnoreCase(phone.toString())) {
+                isPhoneAvailable = true;
+                break;
+            }
+        }
+
+        json responseMessage;
+        // If the requested phone is available, then add the order to the 'OrderQueue'
+        if (isPhoneAvailable) {
+            var phoneDeliverDetails = check <json>newDeliver;
+            // Create a JMS message
+
+            jms:Message queueMessage2 = check jmsSession.createTextMessage(phoneDeliverDetails.toString());
+
+            log:printInfo("order Delivery details  added to the delivery  Queue; CustomerName: '" + newDeliver.customerName +
+                    "', OrderedPhone: '" + newDeliver.deliveryPhoneName + "';");
+
+            // Send the message to the JMS queue
+            _ = jmsProducer2 -> send(queueMessage2);
+
+            // Construct a success message for the response
+            responseMessage = {"Message":"Your order is successfully placed. Ordered phone will be delivered soon"};
+
+        }
+        else {
+            // If phone is not available, construct a proper response message to notify user
+            responseMessage = {"Message":"Requested phone not available"};
+        }
+
+        // Send response to the user
+        response.setJsonPayload(responseMessage);
+        _ = caller -> respond(response);
+    }
+}
+
+jms:Connection conn = new({
+        initialContextFactory: "org.apache.activemq.jndi.ActiveMQInitialContextFactory",
+        providerUrl: "tcp://localhost:61616"
+    });
+
+// Initialize a JMS session on top of the created connection
+jms:Session jmsSession2 = new(conn, {
+        // Optional property. Defaults to AUTO_ACKNOWLEDGE
+        acknowledgementMode: "AUTO_ACKNOWLEDGE"
+    });
+
+// Initialize a queue receiver using the created session
+
+endpoint jms:QueueReceiver jmsConsumer2 {
+    session:jmsSession2,
+    queueName:"DeliveryQueue"
+};
+
+service<jms:Consumer> deliverySystem bind jmsConsumer2 {
+    // Triggered whenever an order is added to the 'OrderQueue'
+    onMessage(endpoint consumer, jms:Message message2) {
+        log:printInfo("New order successfilly received from the Delivery Queue");
+        // Retrieve the string payload using native function
+        string stringPayload2 = check message2.getTextMessageContent();
+        log:printInfo("Delivery Details: " + stringPayload2);
+
+        log:printInfo(" Delivery Details sent to the customer successfully");
```

Comment:
do we need all these logs here? can't we combine maybe 214 and 216?

Also, typo in line 211, "successf**i**lly"

Reply:
done